cicd/2-cicd/deploy-cicd.sh

```diff
@@ -2,14 +2,20 @@
 
 echo Deploying Javabuilder CICD Pipeline
 
-# Create/Update the Javabuilder build/deploy pipeline stack. This is manually created and maintained, but should not require elevated permissions. 
+# Create/Update the Javabuilder build/deploy pipeline stack. This is manually created and maintained, but should not require elevated permissions.
+# Options include:
+# - TARGET_BRANCH: Defaults to `main`, passed as a Parameter for "cicd/2-cicd/cicd.template.yml"
+# - MODE: Can be 'standard' (default) or 'adhoc', passed as a Parameter for "cicd/2-cicd/cicd.template.yml"
+# - GITHUB_BADGE_ENABLED: defaults to true, passed as a Parameter for "cicd/2-cicd/cicd.template.yml"
+# - STACK_NAME: defaults to an automated naming convention, no current use cases for changing this.
```

Comment:
Are adhocs named by `TARGET_BRANCH` or `STACK_NAME`?

Reply:
I think I'm going to get rid of STACK_NAME because it would be safer to just let the script generate the stack name. STACK_NAME in this case is solely referring to the CI/CD resources stack, and not the application resources stack.

If TARGET_BRANCH is 'frog', then this script will create a stack called "javabuilder-frog-cicd" which defines a Pipeline that will create "javabuilder-frog-test" and "javabuilder-frog" stacks.

(there is some logic to handle 'main', which omits the branch from the stacks, if the branch is 'main' you just get 'javabuilder-cicd', 'javabuilder-test', and 'javabuilder' stacks)

Reply:
cleaned this up.